### PR TITLE
RPC tests stdio repair

### PR DESCRIPTION
**Pull request by awemany**
This is a prerequisite for #1160 which I will now rebase on top of this.

### CODE REVIEW

**Comment by awemany** [comment]
Travis truly is a mistery. This time, it fails because it got a `KeyboardInterrupt` (!)

**Comment by ptschip** [comment]
keyboardinterrupt means your rpc-tests.py script didn't recieve any
communication from the subprocess for 20 mins (see line 455
rpc-tests.py). When that happens rpc-tests.py will send a
keyboardinterrupt to end your test and then you get a printout of the
last part of the debug.log.  Looks to me like one of your nodes might
have segfaulted since the last log item wasn't printed out entirely...

**Comment by awemany** [comment]
@ptschip: Ah, thank you, there is a SIGINT and not a SIGTERM indeed. So in that way it makes sense.

But so far the travis timeout is 'terminate a test case after 20min' instead of 'terminate 20min after last communication'. (Because there was only a check for a terminating process and NOT a continuous polling of the communications, which is exactly what these commits change).

Should I change that behavior as well, so that travis terminates only if there's 20min of no comm?

It seems to me that a 20min timeout for a single test should be plenty and the other behavior has the risk of hanging indefinitely.

**Comment by ptschip** [comment]
yes you're right , it is after 20 mins...seems to work fine as is, not
sure it's worth changing.

**Comment by awemany** [comment]
Ok, thanks, then the commit should be fine as is. I don't get why travis fails, seems spurious and as if `listtransactions.py` took forever this time.

File: qa/pull-tester/rpc-tests.py [hunk 1]
```diff
@@ -435,27 +438,45 @@ def get_next(self):
...
+            log_stdout = tempfile.SpooledTemporaryFile(max_size=2**16, mode="w+")
+            log_stderr = tempfile.SpooledTemporaryFile(max_size=2**16, mode="w+")
             self.jobs.append((t,
                               time.time(),
                               subprocess.Popen((RPC_TESTS_DIR + t).split() + self.flags.split() + port_seed,
                                                universal_newlines=True,
                                                stdout=subprocess.PIPE,
-                                               stderr=subprocess.PIPE)))
+                                               stderr=subprocess.PIPE),
+                              log_stdout, log_stderr))
```
Comment (sickpig) [inline qa/pull-tester/rpc-tests.py]: Indent

Comment (awemany) [reply qa/pull-tester/rpc-tests.py]: Should be fine? `log_stdout`, `log_stderr` are part of the tuple appended to the `jobs` list. They are not part of the `subprocess.Popen()`. Or do you mean something else?

Comment (sickpig) [reply qa/pull-tester/rpc-tests.py]: scratch that, you are indeed right.

**Review by sickpig — APPROVED** [review]
great improvements.

**Comment by awemany** [comment]
Thanks! There was an issue with communicate() always timing out, even though it finished succesfully - see updated commit message for details.

**Comment by gandrewstone** [comment]
I think I pulled this in by merging 1160. Please close if there this PR is a subset of 1160

**Comment by ptschip** [comment]
needs rebase

**Comment by sickpig** [comment]
closing. merged via #1160.
(@awemany feel free to reopen in case you think #1168 is not a strict subset of 1160)

**Pull request closed without merging**